### PR TITLE
File related opcodes updates

### PR DESCRIPTION
**Pull request by MiranDMC**
Moved all file io stream opcodes to filesystem plugin.
New opcode read_block_from_file
New opcode get_file_position
Multiple bug fixes, parameters validation and error messages.
Implemented legacy mode for all file streams related opcodes.
Added test script testing all stream opcodes.

```
2300=2,get_file_position %1d% store_to %2d%
2301=3,read_block_from_file %1d% size %2d% buffer %3d%
```

### CODE REVIEW

File: source/CCustomOpcodeSystem.cpp [hunk 1]
```diff
@@ -3098,6 +2665,23 @@ extern "C"
...
 
+	void WINAPI CLEO_ReadStringParamWriteBuffer(CLEO::CRunningScript* thread, char** outBuf, int* outBufSize, DWORD* outNeedsTerminator)
+	{
+		if (thread == nullptr || 
+			outBuf == nullptr || 
+			outBufSize == nullptr ||
+			outNeedsTerminator == nullptr)
+		{
+			LOG_WARNING(thread, "Invalid argument of CLEO_ReadStringParamWriteBuffer in script %s\nScript suspended.", ((CCustomScript*)thread)->GetInfoStr().c_str());
+			return;
```
Comment (x87) [inline source/CCustomOpcodeSystem.cpp:2676]: warning says "script suspended" but it does not happen?

File: cleo_plugins/FileSystemOperations/FileSystemOperations.cpp [hunk 2]
```diff
@@ -58,6 +85,94 @@ class FileSystemOperations
...
+        m_hFiles.insert(handle);
+        CLEO_SetIntOpcodeParam(thread, handle);
+        CLEO_SetThreadCondResult(thread, true);
+        return OR_CONTINUE;
+    }
+
+    //0A9B=1,closefile %1d%
+    static OpcodeResult WINAPI opcode_0A9B(CRunningScript* thread)
+    {
+        DWORD handle = CLEO_GetIntOpcodeParam(thread); OPCODE_VALIDATE_HANDLE(handle)
```
Comment (x87) [inline cleo_plugins/FileSystemOperations/FileSystemOperations.cpp:132]: this probably can be extracted to another macro or method: 

DWORD handle = GetFileHandle(thread); (calls CLEO_GetIntOpcodeParam & does OPCODE_VALIDATE_HANDLE)

File: source/CCustomOpcodeSystem.cpp (hunk 1 above)
Comment (MiranDMC) [reply source/CCustomOpcodeSystem.cpp:2676]: True! Fixed

File: cleo_plugins/FileSystemOperations/FileSystemOperations.cpp (hunk 2 above)
Comment (MiranDMC) [reply cleo_plugins/FileSystemOperations/FileSystemOperations.cpp:132]: Method no, as it uses macro so can call "return" of the nesting fiction.
Maybe it is not good idea to over engineering it. Macros are dirty and do not play nicely when debugging.

Comment (x87) [reply cleo_plugins/FileSystemOperations/FileSystemOperations.cpp:132]: yea I mean you use the same pair of commands 12 times, could extract into a small function to reduce code duplicating

**Review by x87 — APPROVED** [review]

**Pull request merged**